### ISSUE TITLE
WatchOS Telegram app cannot connect under cellular data or without the paired iPhone.

### ISSUE THREAD

**Issue by manudroid19** [TelegramMessenger/Telegram-iOS#415]
### Checklist
- [x] I am reporting an issue in existing functionality that does not work as intended
- [x] I've searched for existing [GitHub issues](https://github.com/telegrammessenger/Telegram-iOS/issues)

### Description
Take a GPS + Cellular Apple Watch. Using Bluetooth or Wi-Fi, telegram app works as expected. However, if the watch switches to cellular data or the iPhone is not near, the Telegram app stops working. 

Also, using a GPS Apple Watch, you can see that if you turn bluetooth off and iPhones Wifi off, connection does not work either.

This is because the file:

https://github.com/TelegramMessenger/Telegram-iOS/blob/d1641281452cd521148ba82d14dc878d7c8ae498/submodules/WatchBridge/Impl/Sources/TGBridgeServer.m#L4

This class has not been updated for years, and it's using the old school Watch Connectivity framework, that routes all connectivity throught the companion iOS app.

The good way to implement this is using NSURLSession class, which is far more modern and robust. If the iPhone is avaliable, it will route the traffic via bluetooth, but if it isn't, watchOS can communicate with Telegram on its own.

PLEASE consider implementing this feature. It should not be diffcult (NSURLSession is already used in the iOS app to communicate with Telegram), and there's a lot of people that loves leaving their phones and using just their watch to communicate. This would also give an strategical advantage over WhatsApp, that doesn't even have a watch app.

### Expected Behavior
Telegram for watchOS should open normally.

### Actual Behavior
Telegram for watchOS cannot connect.

### Steps to Reproduce
Using GPS+Cellular over cellular:
1. Turn off bluetooth on the watch
2. Turn off wifi on the watch
3. Open Telegram app on the watch

Using GPS+Cellular or GPS over Wi-fi:
1. Turn off bluetooth on the watch
2. Turn off wifi on the iPhone
3. Open Telegram app on the watch

### Screenshots and Videos 
![IMG_0279(2)](https://user-images.githubusercontent.com/6536835/98988081-67c75100-2527-11eb-96f4-7f57189685e1.PNG)

### Environment
**Device:** `iPhone 12, Watch Series 5 Cellular`

**iOS version**: `14.2`

**App version:** `7.2`

**Comment by a6unraj** [comment]
I was also looking for using Telegram standlaone in my cellular version watch. Was this working before ?

**Comment by manueldeprada** [comment]
> I was also looking for using Telegram standlaone in my cellular version watch. Was this working before ?

please vote here to have more chances devs will implement it:

https://bugs.telegram.org/c/76

**Comment by SebDominguez** [comment]
I was really surprised to came across this issue with my new Apple Watch. I hope this will get fixed soon

**Comment by emanuelelaface** [comment]
I voted in the bugs list of Telegram but I think they will not fix it anytime soon because the app depends on the phone for many things. For example the authentication at the beginning, then on the attachment there is the button "View on the phone" etc. It is really a pity, so far I found only FB Messenger as independent app for chat, that is probably the worst chat app around. I don't know why developers are so distant from wearable devices.

**Comment by gracjankn** [comment]
> > I was also looking for using Telegram standlaone in my cellular version watch. Was this working before ?
> 
> please vote and comment here to have more chances devs will implement it:
> 
> https://bugs.telegram.org/c/76

I just voted there now. I'm buying an  Watch in a few months, and I'm very disappointed to hear about the problems Telegram has on that platform. For many people the main point of a smartwatch is that you don't need your phone with you all the time. When going for a walk, I'd just leave my phone at home. Telegram is probably the only app I use now that would undermine this premise.

There's such a disconnect between the praise Telegram gets on pretty much every operating system on the planet and the criticism it gets on watchOS and Wear OS.

**Comment by gracjankn** [comment]
There's hope. I find out Telegram has announced a [coding contest](https://t.me/contest/220) with a task of creating a standalone Watch app and deadline on 23:59 on March, 21 (Dubai time). After the initial announcement message (what I linked here), there was a second one with more details of how it should be done, followed by mockups in PNG and Sketch file formats.

**Comment by emanuelelaface** [comment]
Oh, this is the best answer to this thread. I will follow this contest then and see what will be the outcome. Thanks!

**Comment by gracjankn** [comment]
> Oh, this is the best answer to this thread. I will follow this contest then and see what will be the outcome. Thanks!

You're very welcome! 👍🏻☀️

I'm delighted that in a span of a few hours I went from _oh there are all these problems with the Telegram watch app, it's probably gonna be complete trash because everyone is complaining about it being buggy and lacking basic features, and it crushes life-changing dreams about the way I'd use technology after getting a watch_ to _actually there are already people working on it, we're very close to the first related deadline, and it'll be 10 times better than I'd ever hope for since it's a complete rewrite!_

What a rollercoaster…

**Comment by timvdsm** [comment]
This would be absolutely great to be able to use it standalone with an Apple Watch cellular. Is there any update?

**Comment by gracjankn** [comment]
> Is there any update?

[The last message about watchOS app contest so far](https://t.me/contest/237) is from March 25th, saying they received 9 submissions from contestants for the Round 2.

We have to wait for new information. They can either announce final winners of the entire contest or announce winners of the Round 2 and decide there's gonna be a Round 3. We don't know yet.

**Comment by emanuelelaface** [comment]
The problem is that the TDLib was not compiling for real devices but just for the simulators, so they were not able to evaluate the submissions. Since then the lib was changed on the repository but no news was added. It is now more than one month without any news. It is really a pity because I was expecting something and the blocking part seems from the Telegram side, not from who sent the solutions.

**Comment by gracjankn** [comment]
> the blocking part seems from the Telegram side

Yes, totally looks like it. I'm buying my first  Watch this month and I'd like to know how long I'm gonna have to wait for the Telegram to catch up with the rest of the system. I won't be able to fully enjoy my new device, until my favorite messaging app does no longer suck on it, yet there haven't been any news for over a month.

I don't have to know who won right away or an exact ETA when the official version will arrive or anything like that. I understand the development job is hard, but I'd like them to at least tell something. At least assure us that some progress is being made, but something is more complicated and will require more time than they thought, instead announcing a contest, taking submission and leaving us wondering if they even remember.

**Comment by graypegg** [comment]
Round 3 has been posted! https://t.me/contest/258

**Comment by brandoncote** [comment]
> This would be absolutely great to be able to use it standalone with an Apple Watch cellular. Is there any update?

the developer of FlickType has just released a standalone version of telegram for watchOS! it's called Nanogram and available at https://t.me/NanogramApp/5

**Comment by iMonZ** [comment]
Anything new here?

**Comment by tukaaa** [comment]
+1 to this issue

**Comment by bpasero** [comment]
+1, would love to see this happening. I trade a VSCode fix for this fix 👍
